internal/build.go

```diff
@@ -75,3 +82,31 @@ func buildCommandLoaderFilter(bundleMode bool, ignore []string) func(string, os.
 		return util.LoaderFilter{Ignore: ignore}.Apply(abspath, info, depth)
 	}
 }
+
+func assertUniqueRuleIds(paths []string) error {
+	rules, err := util.RetrieveRules(paths)
+	if err != nil {
+		return err
+	}
+
+	pathsByVisitedRuleIds := make(map[string]string)
```

Comment:
nit: `pathsByVisitedRuleIds` confused me a bit (although I might have holiday brain). I'm not sure about this one either, but what do you think of `visitedRulePaths`? It technically contains the paths of all visited rules, and we can see from the type that it's a map